### PR TITLE
br: use schema diff reload after log restore

### LINKED ISSUE #58398: br: add option to reload schema in diff rather than full reload after log restore
## Enhancement

Full reload takes too much time if there are millions table, it's fine for restoring to a fresh cluster but if we want to have online restore it can impact traffic. Use diff reload instead

### PR DESCRIPTION
**Pull request by Tristan1900**
### What problem does this PR solve?

Issue Number: close #58398

Problem Summary:

### What changed and how does it work?
Instead of doing full reload after log restore, we added another diff reload command `logRestore` action to diff reload changes to reduce the impact to online traffic.

1. after log restore we have all the affected tables ids stored in the table replace, we can pass these ids to the info schema 
2. info schema need to bridge the gap between what's stored in TiKV and what's in the info schema, it checks the difference and either drop the schema/table or create/update schema/table.
3. to further reduce the impact that might bring to the online traffic, we can apply the diff change in batches. We can assume queries against log restored tables are not consistent during log restore as it is now, so apply changes in batch will not be a new issue

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/infoschema/builder.go [hunk 1]
```diff
@@ -101,6 +101,8 @@ func (b *Builder) ApplyDiff(m meta.Reader, diff *model.SchemaDiff) ([]int64, err
 		return applyExchangeTablePartition(b, m, diff)
 	case model.ActionFlashbackCluster:
 		return []int64{-1}, nil
+	case model.ActionLogRestore:
+		return applyLogRestore(b, m, diff)
```
Comment (D3Hunter) [inline pkg/infoschema/builder.go:105]: this doesn't follow online schema change. it's possible some node doesn't load the diff in time, and breaks linearizability

**Comment by Tristan1900** [comment]
closed as discusses this way of doing is a bit hacky, need to use a new feature that DDL is currently developing to make changes go through DDL

**Pull request closed without merging**